src/zenml/models/v2/core/deployment.py

```diff
@@ -0,0 +1,315 @@
+
+
+# ------------------ Request Model ------------------
+
+
+class DeploymentRequest(ProjectScopedRequest):
+    """Request model for deployments."""
+
+    name: str = Field(
+        title="The name of the deployment.",
+        description="A unique name for the deployment within the project.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    snapshot_id: UUID = Field(
+        title="The pipeline snapshot ID.",
+        description="The ID of the pipeline snapshot associated with the "
+        "deployment.",
+    )
+    deployer_id: UUID = Field(
+        title="The deployer ID.",
+        description="The ID of the deployer component managing this deployment.",
+    )
+    auth_key: Optional[str] = Field(
+        default=None,
+        title="The auth key of the deployment.",
+        description="The auth key of the deployment.",
+    )
+
+
+# ------------------ Update Model ------------------
+
+
+class DeploymentUpdate(BaseUpdate):
+    """Update model for deployments."""
+
+    name: Optional[str] = Field(
+        default=None,
+        title="The new name of the deployment.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    snapshot_id: Optional[UUID] = Field(
+        default=None,
+        title="New pipeline snapshot ID.",
+    )
+    url: Optional[str] = Field(
+        default=None,
+        title="The new URL of the deployment.",
+    )
+    status: Optional[str] = Field(
+        default=None,
+        title="The new status of the deployment.",
+    )
+    deployment_metadata: Optional[Dict[str, Any]] = Field(
+        default=None,
+        title="The new metadata of the deployment.",
+    )
+    auth_key: Optional[str] = Field(
+        default=None,
+        title="The new auth key of the deployment.",
+    )
+
+    @classmethod
+    def from_operational_state(
+        cls, operational_state: DeploymentOperationalState
+    ) -> "DeploymentUpdate":
+        """Create an update from an operational state.
+
+        Args:
+            operational_state: The operational state to create an update from.
+
+        Returns:
+            The update.
+        """
+        return cls(
+            status=operational_state.status,
+            url=operational_state.url,
+            deployment_metadata=operational_state.metadata,
+        )
+
+
+# ------------------ Response Model ------------------
+
+
+class DeploymentResponseBody(ProjectScopedResponseBody):
+    """Response body for deployments."""
+
+    url: Optional[str] = Field(
+        default=None,
+        title="The URL of the deployment.",
+        description="The HTTP URL where the deployment can be accessed.",
+    )
+    status: Optional[str] = Field(
+        default=None,
+        title="The status of the deployment.",
+        description="Current operational status of the deployment.",
+    )
+
+
+class DeploymentResponseMetadata(ProjectScopedResponseMetadata):
+    """Response metadata for deployments."""
+
+    snapshot_id: Optional[UUID] = Field(
```

Comment:
Both the snapshot and the deployer are already included in the resources, which makes these ids unnecessary I think? In our current hydration logic, I don't think it's possible to fetch a response with metadata but without resources, so this will always be duplicate information.